catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroMyProjectsActivityTest.java

```diff
@@ -0,0 +1,126 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.activity;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.ui.MyProjectsActivity;
+import org.catrobat.catroid.ui.fragment.ProjectListFragment;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.ProjectListDataInteractionWrapper.onProjectWithName;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroMyProjectsActivityTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<MyProjectsActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(MyProjectsActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		deletePhiroPrograms();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroProjectsInitiallyLoadedTest() {
+		for (String projectName : ProjectListFragment.PHIRO_PROJECTS) {
+			onProjectWithName(projectName)
+					.check(matches(isDisplayed()));
+		}
+	}
+
+	@Test
+	public void phiroProgramRestoredWhenDeletedTest() {
+		String projectName = ProjectListFragment.PHIRO_PROJECTS[0];
+		StorageHandler.getInstance().deleteProject(projectName);
+
+		relaunchActivity();
+
+		onProjectWithName(projectName)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroProgramsUnchangedWhenModifiedTest() throws InterruptedException {
+		addNoteBrickToFirstScript(ProjectListFragment.PHIRO_PROJECTS[0]);
+
+		relaunchActivity();
+
+		checkAddedNoteBrickIsStillInFirstScript();
+	}
+
+	private void relaunchActivity() {
+		baseActivityTestRule.getActivity().finish();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	private void addNoteBrickToFirstScript(String projectName) {
```

Comment:
ok, so heres the run on your repository:
https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/19319/

and here the one on the feature/Nolb
https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/19320/

since most of the tests are in another package, they arent run here in the first place...
weve got to think up a proper way on how to deal with the flavored tests i guess...